From 80c5549318cec082da0793526e8e471e946103be Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Tue, 21 Feb 2017 22:25:04 +0100
Subject: [PATCH] iplink: bridge_slave: add support for
 IFLA_BRPORT_MULTICAST_ROUTER

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1374360
Upstream Status: iproute2.git commit 10759a90ab589
Conflicts: Context changes due to missing commits:
* f6e615dec9ee9 ("iplink: bridge_slave: add support for
                  IFLA_BRPORT_PROXYARP")
* 38b31a78da859 ("iplink: bridge_slave: add support for
                  IFLA_BRPORT_PROXYARP_WIFI")

commit 10759a90ab5896b5b0ca8ddd7b91c5b3a7466062
Author: Nikolay Aleksandrov <nikolay@cumulusnetworks.com>
Date:   Tue Feb 16 16:08:54 2016 +0100

    iplink: bridge_slave: add support for IFLA_BRPORT_MULTICAST_ROUTER

    Add support to be able to view and change IFLA_BRPORT_MULTICAST_ROUTER port
    attribute.

    Signed-off-by: Nikolay Aleksandrov <nikolay@cumulusnetworks.com>
---
 ip/iplink_bridge_slave.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/ip/iplink_bridge_slave.c b/ip/iplink_bridge_slave.c
index 4593872..9ed44b1 100644
--- a/ip/iplink_bridge_slave.c
+++ b/ip/iplink_bridge_slave.c
@@ -29,6 +29,7 @@ static void print_explain(FILE *f)
 		"                        [ root_block {on | off} ]\n"
 		"                        [ learning {on | off} ]\n"
 		"                        [ flood {on | off} ]\n"
+		"                        [ mcast_router MULTICAST_ROUTER ]\n"
 	);
 }
 
@@ -98,6 +99,10 @@ static void bridge_slave_print_opt(struct link_util *lu, FILE *f,
 	if (tb[IFLA_BRPORT_UNICAST_FLOOD])
 		print_onoff(f, "flood",
 			rta_getattr_u8(tb[IFLA_BRPORT_UNICAST_FLOOD]));
+
+	if (tb[IFLA_BRPORT_MULTICAST_ROUTER])
+		fprintf(f, "mcast_router %u ",
+			rta_getattr_u8(tb[IFLA_BRPORT_MULTICAST_ROUTER]));
 }
 
 static void bridge_slave_parse_on_off(char *arg_name, char *arg_val,
@@ -162,6 +167,14 @@ static int bridge_slave_parse_opt(struct link_util *lu, int argc, char **argv,
 			NEXT_ARG();
 			bridge_slave_parse_on_off("flood", *argv, n,
 						  IFLA_BRPORT_UNICAST_FLOOD);
+		} else if (matches(*argv, "mcast_router") == 0) {
+			__u8 mcast_router;
+
+			NEXT_ARG();
+			if (get_u8(&mcast_router, *argv, 0))
+				invarg("invalid mcast_router", *argv);
+			addattr8(n, 1024, IFLA_BRPORT_MULTICAST_ROUTER,
+				 mcast_router);
 		} else if (matches(*argv, "help") == 0) {
 			explain();
 			return -1;
-- 
1.8.3.1

